synapse/handlers/register.py

```diff
@@ -414,6 +415,30 @@ def _join_user_to_room(self, requester, room_identifier):
             ratelimit=False,
         )
 
+    def check_registration_ratelimit(self, address):
+        """A simple helper method to check whether the registration rate limit has been hit
+        for a given IP address
+
+        Args:
+            address (str): the IP address used to perform the registration.
+
+        Raises:
+            LimitExceededError: If the rate limit has been exceeded.
+        """
+        time_now = self.clock.time()
+
+        allowed, time_allowed = self.ratelimiter.can_do_action(
```

Comment:
```suggestion
        self.ratelimiter.ratelimit(
```

which will correctly raise `LimitExceededError` itself